hello_world_on_chain/test/user.ts

```diff
@@ -1,79 +1,95 @@
 
-        const taBytes = ethers.decodeBase64(data.transitive_attested_function_call.transitive_attestation)
-        const ta = Buffer.from(taBytes).toString('utf-8');
+        const taBytes: Uint8Array = ethers.decodeBase64(
+            data.transitive_attested_function_call.transitive_attestation
+        )
+        const ta: string = Buffer.from(taBytes).toString('utf-8');
 
-        return {
-            publicKey: `0x${publicKeyHex}`,
-            transitiveAttestation: ta
-        };
+        return {publicKey: `0x${publicKeyHex}`, transitiveAttestation: ta};
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`Error loading EVM link data: ` + e);
     }
 }
 
-const loadUserDeployedAddress = () => {
+function loadUserDeployedAddress(): string {
     try {
-        const dir = path.resolve(
+        const dir: string = path.resolve(
             __dirname,
             "../deployments/user_deployed_address"
         )
-        const file = fs.readFileSync(dir, "utf8")
+        const file: string = fs.readFileSync(dir, "utf8")
 
         return file.toString()
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`loading user deployed address: ` + e);
     }
 }
 
-const loadUserContractABI = () => {
+function loadUserContractABI(): any {
     try {
-        const dir = path.resolve(
+        const dir: string = path.resolve(
             __dirname,
             "../artifacts/contracts/User.sol/User.json"
         )
-        const file = fs.readFileSync(dir, "utf8")
-        const json = JSON.parse(file)
+        const file: string = fs.readFileSync(dir, "utf8")
+        const json: any = JSON.parse(file)
         return json.abi
     } catch (e) {
-        console.log(`e`, e)
+        throw new Error(`loading user contract ABI: ` + e);
     }
 }
 
-describe("Local Test", function () {
-    async function deployUser() {
-        const contract = await hre.ethers.deployContract("User");
+
+interface UserContract extends ethers.Contract {
+    // @ts-ignore
+    processTransitivelyAttestedHelloWorldOutput(publicKey: string, ta: string): Promise<ethers.ContractTransactionResponse>;
+}
+
+describe("Local Test", function (): void {
+    async function deployUser(): Promise<{ userContract: User }> {
+        const contract: User = await hre.ethers.deployContract("User");
         return {userContract: contract};
     }
 
-    it("Set signing key and verify TA", async () => {
+    it("Verify TA", async (): Promise<void> => {
         // given
-        const evmLinkData = loadEVMLinkData("../inputs/out.json");
-        const publicKey = evmLinkData.publicKey;
+        const evmLinkData: EVMLinkData = loadEVMLinkData("../inputs/out.json");
+        const publicKey: string = evmLinkData.publicKey;
 
-        const {userContract} = await loadFixture(deployUser);
-        await userContract.setTASigningKeyAddress(publicKey as any);
+        const {userContract} = await loadFixture(deployUser) as UserContract;
```

Comment:
I added the `as UserContract` so that we can use the defined interface and don't have to add `as any` on line 121